frappe/utils/file_manager.py

```diff
@@ -450,4 +450,32 @@ def add_attachments(doctype, name, attachments):
 			f = save_url(attach.file_url, attach.file_name, doctype, name, "Home/Attachments", attach.is_private)
 			files.append(f)
 
-	return files
\ No newline at end of file
+	return files
+
+@frappe.whitelist()
+def download_zip(files, output_filename, public=False):
+	from werkzeug.exceptions import Forbidden
+	from zipfile import ZipFile
+
+	output_folder = 'public'
+	if not public:
+		try:
+			frappe.only_for(("System Manager", "Accounts Manager", "Administrator"))
```

Comment:
can't hardcode, check the permission on the individual files

Reply:
File not being created via File doctype. Should it be?